library/src/main/java/com/pokegoapi/api/settings/AsyncCatchOptions.java

```diff
@@ -151,23 +161,22 @@ public Pokeball getItemBall(double encounterProbability) throws LoginFailedExcep
 			if (bag.getItem(pokeBall.getBallType()).getCount() > 0) {
 				return pokeBall;
 			} else if (useBestPokeball) {
-				if (!skipMasterBall) {
-					if (bag.getItem(ITEM_MASTER_BALL).getCount() > 0) {
-						return MASTERBALL;
-					}
+				if (!skipMasterBall && bag.getItem(ITEM_MASTER_BALL).getCount() > 0) {
+					return MASTERBALL;
 				} else if (bag.getItem(ITEM_ULTRA_BALL).getCount() > 0) {
 					return ULTRABALL;
+				} else if (bag.getItem(ITEM_GREAT_BALL).getCount() > 0) {
+					return GREATBALL;
 				}
-			} else if (bag.getItem(ITEM_POKE_BALL).getCount() > 0) {
+			}
+			if (bag.getItem(ITEM_POKE_BALL).getCount() > 0) {
 				return POKEBALL;
 			}
 			throw new NoSuchItemException("No ball available for capture probability");
 		} else {
-			int index = 3;
-			if (pokeBall.getBallType() == ITEM_MASTER_BALL) index = 3;
-			if (pokeBall.getBallType() == ITEM_ULTRA_BALL) index = 2;
-			if (pokeBall.getBallType() == ITEM_GREAT_BALL) index = 1;
-			if (pokeBall.getBallType() == ITEM_POKE_BALL) index = 0;
+			ItemId[] pokeballs = new ItemId[] { ITEM_MASTER_BALL, ITEM_ULTRA_BALL,
+					ITEM_GREAT_BALL, ITEM_POKE_BALL };
```

Comment:
Then you had to chain .add commands together. I thought you were all about readability?